src/main/java/org/hydev/mcpm/utils/ColorLogger.java

```diff
@@ -0,0 +1,98 @@
+package org.hydev.mcpm.utils;
+
+import org.bukkit.command.CommandSender;
+
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import static org.hydev.mcpm.utils.Sugar.uncheckedMapOf;
+
+/**
+ * Logging with minecraft color codes (see https://minecraft.fandom.com/wiki/Formatting_codes)
+ *
+ * @author Azalea (https://github.com/hykilpikonna)
+ * @since 2022-11-20
+ */
+public class ColorLogger
+{
+    private static final Map<String, String> ansiReplacements = uncheckedMapOf(
+        "&0", "\033[0;30m",
+        "&1", "\033[0;34m",
+        "&2", "\033[0;32m",
+        "&3", "\033[0;36m",
+        "&4", "\033[0;31m",
+        "&5", "\033[0;35m",
+        "&6", "\033[0;33m",
+        "&7", "\033[0;37m",
+        "&8", "\033[1;30m",
+        "&9", "\033[1;34m",
+        "&a", "\033[1;32m",
+        "&b", "\033[1;36m",
+        "&c", "\033[1;31m",
+        "&d", "\033[1;35m",
+        "&e", "\033[1;33m",
+        "&f", "\033[1;37m",
+        "&r", "\033[0m",
+        "&l", "\033[1m",
+        "&o", "\033[3m",
+        "&n", "\033[4m",
+        "&-", "\n"
```

Comment:
Also do we have a logging library that can handle this for us? Logging libraries should be responsible to handle escape sequences in a proper way. Some terminals (notably Xcode) don't support these escapes and the result is a bunch of garbled text.

Reply:
Check out https://github.com/HyDevelop/HyLogger (I made it lol)

Also [HyFetch can try to detect](https://github.com/hykilpikonna/hyfetch/blob/b86f43e9abc6e7b618eedb06d889fe404d55b886/hyfetch/termenv.py#L18) if a terminal supports Xterm256 or Xterm truecolor but not if it doesn't support ANSI.

But tbh there's really no way of correctly identifying if a terminal supports ANSI that works 100% of the time. There are so many different terminals out there and not all of them follow the same standard.

Reply:
I'll just assume that whoever is using the tool on the command line is using it with a terminal that supports colors.

Reply:
> There are so many different terminals out there and not all of them follow the same standard.

This is why we move the responsibility to a library maintained by other people. If a terminal is not supported, then it's their fault to support it. As developers we can't possibly think of all the different environments our app will run in, but the library developers can do that thinking for us.

Again not a blocker. It's the last day before feedback lol

Reply:
Quick fix? 🤔 

```java
var ampEscape = "[gFk38N-EsCaPeD AmPeRcEnT-f47Svw]";
in = in.replace("&&", ampEscape);

// Replace colors

in = in.replace(ampEscape, "&");
```

Reply:
> If a terminal is not supported, then it's their fault to support it. 

when you use intuition to skip steps when showing work and lose marks:
"it's not my fault you need more steps"

Reply:
Man, this is a library. It's not like omitting steps 3 through 5, it's like citing a proven theorem.